systemverilog-plugin/UhdmAst.cc

```diff
@@ -1088,7 +1089,22 @@ AST::AstNode *UhdmAst::process_value(vpiHandle obj_h)
         // Surelog reports constant integers as a unsigned, but by default int is signed
         // so we are treating here UInt in the same way as if they would be Int
         case vpiUIntVal:
+            if (val.value.uint > std::numeric_limits<std::uint32_t>::max()) {
+                // an integer is by default signed, so use 'sd despite the variant vpiUIntVal
+                strValType = "'sd";
+                string str_value = std::to_string(val.value.uint);
+                val.value.str = strdup(str_value.c_str());
+                break;
+            }
+            [[fallthrough]];
         case vpiIntVal: {
+            if (val.value.integer > std::numeric_limits<std::int32_t>::max()) {
+                strValType = "'sd";
+                string str_value = std::to_string(val.value.integer);
+                val.value.str = strdup(str_value.c_str());
+                break;
+            }
+
```

Comment:
Later we are always using ``val.value.integer`` (https://github.com/chipsalliance/yosys-f4pga-plugins/blob/main/systemverilog-plugin/UhdmAst.cc#L1113) for both ``vpiUIntVal`` and ``vpiIntVal``. Maybe we can simplify this to the single case, or we should fix usage of ``val.value.integer`` in case of ``vpiUIntVal``

Reply:
Alright, I've changed this so that now the proper variant from `val.value` is used in the call to `mkconst_int`.